File: project10/ruby/birds.rb
# birds.rb | Tests out the menagerie of bird classes.
#
# Completed by: Haim Hong
# Date: March 25 2022
######################################################

# to have all these classes to this file and to be used
require './Bird.rb'
require './Duck.rb'
require './Goose.rb'
require './Owl.rb'
require './Kiwi.rb'
require './Ostrich.rb'
require './Penguin.rb'

puts "\nWelcome to the Bird Park!\n\n"

bird0 = Bird.new "Hawkeye"
bird0.print

bird1 = Duck.new "Donald"
bird1.print

bird2 = Goose.new "Mother"
bird2.print

bird3 = Owl.new "Woodsey"
bird3.print

bird4 = Kiwi.new "Fruit"
bird4.print

bird5 = Ostrich.new "Orville"
bird5.print

bird6 = Penguin.new "Peter"
bird6.print

puts "\n\n"

File: project10/java/Birds.java
/* Birds.java illustrates inheritance and polymorphism.
 *
 * Begun by: Joel Adams, for CS 214 Lab 10 at Calvin College.
 * Completed by: Haim Hong
 * Date: Marh 25 2022
 *********************************************************/

public class Birds {
  
  public static void main(String[] args) {
      System.out.println("\nWelcome to the Bird Park!\n");
      // testubg each kind of birds
      Bird bird0 = new Bird();

      Bird bird1 = new Bird("Hawkeye");
      bird1.print();

      Duck bird2 = new Duck("Donald");
      bird2.print();

      Goose bird3 = new Goose("Mother Goose");
      bird3.print();

      Owl bird4 = new Owl("Woodsey");
      bird4.print();

      Bird bird5 = new Ostrich("Orville");
      bird5.print();

      Bird bird6 = new Kiwi("Fruit");
      bird6.print();

      Bird bird7 = new Penguin("Peter");
      bird7.print();
      System.out.println();


    }
}

